web/src/components/navbar/NavBarDashboardLink.tsx

```diff
@@ -0,0 +1,28 @@
+import { UnStyledButton } from "@/components/njwds-extended/UnStyledButton";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import { useRouter } from "next/router";
+import { ReactNode } from "react";
+
+interface Props {
+  businessName: string;
+  stringToDisplay: string;
+  className?: string;
+}
+
+export const NavBarDashboardLink = (props: Props): ReactNode => {
+  const router = useRouter();
+  return (
+    <div>
+      <UnStyledButton
+        onClick={(): void => {
+          router.push(ROUTES.dashboard);
+        }}
+        ariaLabel={`My Account for ${props.businessName} Home`}
+      >
+        <span className={`text-primary text-semibold font-body-sm ${props.className ?? ""}`}>
+          {props.stringToDisplay}
+        </span>
+      </UnStyledButton>
+    </div>
+  );
+};
```

Comment:
Looking at this after it's been implemented, it's kindof goofy that we have a link to the dashboard on the dashboard. Might have made more sense for us to only make this a link on tasks/details/profile.

Reply:
I'm not sure I agree, I feel like its a well established pattern to just always hame a home icon in the top right regardless of whether or not you're on the home page.

Reply:
This wouldn't be the home icon, but rather the my account or business name portion which takes you to the dashboard.

Reply:
I made it already that you can't click it to navigate to the dashboard while you are in onboarding so we could make it so it doesn't work on the dashboard page with some more logic if we wanted?

Reply:
[sand] @somabadri If I'm in the "add another business" onboarding flow, and I click the "My Account" link text, I get redirected to the onboarding flow and no longer have the option to return to the dashboard. Can we make sure that essentially has the same behavior as the "Return to ${business_name}" button in that use case?
<img width="313" alt="Screenshot 2024-01-31 at 3 27 01 PM" src="https://github.com/newjersey/navigator.business.nj.gov/assets/22647707/689747b4-caa9-4d1f-b779-46f22b1a7e55">

Reply:
Did design confirm the link should be disabled. I agree with Aaron, there should be a consistent pattern, Conditionally disable/enable a link is a jarring experience.

Reply:
@FarazA22 I asked about this in slack and the consensus was that it should always navigate to dashboard